### DISCUSSION TITLE
Denominators with non-integer powers

### DISCUSSION THREAD

**Discussion by phenomax** [rnlg/LiteRed2#6] [Q&A]
Hi,
is there currently a way (or maybe a workaround) to implement denominators with non-integer, e.g., fractional powers?
Concretely, I am looking for a way to implement a denominator 1/( (p^2)^( alpha/2 )) with alpha an integer
Any help would be highly appreciated!

**Comment by rnlg** [comment]
Unfurtunately, this functionality is not yet functioning properly. It will hopefully be implemented via option `PowerShifts->{a1,a2,...}` of `NewDsBasis`, where a1,a2.. are analytic regularization of the corresponding denominators (some may be zero).

**Reply by phenomax** [reply]
Are there any next steps one could assist you with? 
I tried defining a 

```
NewDsBasis[L2, {k1, k2, p - k2, p - k1, k1 - k2}, {k1, k2}, 
 Directory -> "LiteRed/basis_two_loop", SolvejSector -> True, 
 PowerShifts -> {1/2, 1/2, 1/2, 1/2, 1/2}]
 ```
but received master integrals with negative powers:
``` 
{j[L2, 0, 0, 0, 0, 0], j[L2, -1, 0, 0, 0, 0], j[L2, 0, -1, 0, 0, 0]}
```

**Reply by rnlg** [reply]
This is probably correct result. You should add PowerShifts to the indices, so 
j[L2, 0, 0, 0, 0, 0] actually denotes j[L2, 1/2, 1/2, 1/2, 1/2, 1/2]. Note that this integral is not zero in dim reg.

The real problem is that sometimes SolvejSector would not stop for integrals with shifted powers.

**Reply by phenomax** [reply]
@rnlg Thank you, that was very helpful!

Is there a way to also use fractions for the powers of the integrals? Currently, it seems that a choice must be made whether the powers of a denominator should always be integer or fractional.

In the above example, no denominator with a power of 1 would be allowed as the corresponding index would be 1/2, e.g.
j[L2, 1/2, 0, 0, 0, 0] , then undoing the shift this equals j[L2, 1, 1/2, 1/2, 1/2, 1/2], which is not reducible to the generated master integrals.

**Reply by rnlg** [reply]
@phenomax I see what you mean. But IBP identities relate integrals with **integer** difference between the corresponding indices. So, you can define several bases with different power shifts. Of course, there may be symmetries which relate integrals from different bases, but this is, in principle, the task of `FindExtSymmetries` routine. However, I don't remember if `FindExtSymmetries` correctly accounts power shifts. I will try to check it when I have time. Still, I repeat, the functionality related to `PowerShifts` is not safe enough, so you should be very careful with the results.

**Reply by phenomax** [reply]
Hi @rnlg ,
I believe to have found a basis with half-integer shifts, where SolvejSector does not finish.

```
NewDsBasis[L2a24frac, {p1, p2, q1 - p2, q1 - p1, p1 - p2}, {p1, p2},
  PowerShifts -> {0, 1/2, 0, 1/2, 0}, GenerateIBP -> True,
  FindSymmetries -> False, AnalyzeSectors -> True,
  SolvejSector -> True];
  ```

Do you have any advise or workaround?

Thanks!

**Reply by rnlg** [reply]
There is no solid solution as of now. You can try to change the order of denominators and see if it works.
